src/org/labkey/test/tests/SampleTypeNameExpressionTest.java

```diff
@@ -180,6 +192,71 @@ public void testSimpleNameExpression()
     }
 
+
+    // Issue 50924: LKSM: Importing samples using naming expression referencing parent inputs with # result in error
+    @Test
+    public void testDeriveFromCommentLikeParents()
+    {
+        final String sampleTypeName = "ParentNamesExprTest";
+
+        log("Verify import tsv to create derivative would ignore lines starting with #");
+        String nameExpression = "${MaterialInputs/" + PARENT_SAMPLE_TYPE + "}-child";
+        String data = "MaterialInputs/" + PARENT_SAMPLE_TYPE + "\n";
+        data += PARENT_SAMPLE_01 + "\n";
+        data += PARENT_SAMPLE_01 + "," + PARENT_SAMPLE_02 + "," + PARENT_SAMPLE_03 + "\n";
+        // tsv lines starting with # should be ignored
+        data += PARENT_SAMPLE_03 + "\n";
+        data += PARENT_SAMPLE_03 + "," + PARENT_SAMPLE_02 + "\n";
+
+        SampleTypeHelper sampleHelper = new SampleTypeHelper(this);
+        sampleHelper.createSampleType(new SampleTypeDefinition(sampleTypeName)
+                        .setNameExpression(nameExpression), data);
+
+        DataRegionTable materialTable = new DataRegionTable("Material", this);
+        List<String> names = materialTable.getColumnDataAsText("Name");
+
+        log("generated sample names:");
+        names.forEach(this::log);
+
+        assertEquals(2, names.size());
+
+        assertEquals(PARENT_SAMPLE_01 + "-child", names.get(1));
+        assertEquals("[" + PARENT_SAMPLE_01 + ", " + PARENT_SAMPLE_02 + ", " + PARENT_SAMPLE_03 + "]-child", names.get(0));
+
+        log("Verify import tsv should successfully create derivatives from parent starting with #, as long as this is not the 1st field in the row");
+        data = "Description\tMaterialInputs/" + PARENT_SAMPLE_TYPE + "\n";
+        data += "Parent with leading # should work\t" + PARENT_SAMPLE_03 + "\n";
+        data += "Parents with leading # should work\t" + PARENT_SAMPLE_03 + "," + PARENT_SAMPLE_02 + "\n";
+
+        sampleHelper.bulkImport(data);
+
+        names = materialTable.getColumnDataAsText("Name");
+        log("generated sample names:");
+        names.forEach(this::log);
+
+        assertEquals(4, names.size());
+
+        assertEquals("[" + PARENT_SAMPLE_03 + ", " + PARENT_SAMPLE_02 + "]-child", names.get(0));
+        assertEquals(PARENT_SAMPLE_03 + "-child", names.get(1));
+
+        log("Verify import EXCEL should not ignore lines starting with #");
+        sampleHelper.bulkImport(PARENT_EXCEL);
+
+        names = materialTable.getColumnDataAsText("Name");
+        log("generated sample names:");
+        names.forEach(this::log);
+
+        assertEquals(7, names.size());
+        assertEquals("[" + PARENT_SAMPLE_01 + ", " + PARENT_SAMPLE_04 + "]-child", names.get(0));
+        assertEquals("[" + PARENT_SAMPLE_04 + ", " + PARENT_SAMPLE_03 + "]-child", names.get(1));
+        assertEquals(PARENT_SAMPLE_04 + "-child", names.get(2));
+
+        log("Verify importing tsv to create sample should ignore lines starting with #");
```

Comment:
And TSV should work when #-named sample is not the first field? Or is this covered above and I'm not seeing it.

Reply:
pushed..